http/src/main/scala/org/http4s/blaze/http/http20/ByteBufferInputStream.scala

```diff
@@ -0,0 +1,42 @@
+package org.http4s.blaze.http.http20
+
+import java.io.{IOException, InputStream}
+import java.nio.ByteBuffer
+
+final class ByteBufferInputStream(buffer: ByteBuffer) extends InputStream {
+
+  private var markSize = 0
+
+  override def read(): Int = {
+    markSize -= 1
+    buffer.get()
```

Comment:
Don't you need to do a remaining check?

Also, `buffer.get` returns a signed `byte`, if I remember correctly.  You need to do some bit math to get this into the [0, 256) range.